### ISSUE TITLE
[Request] Break source into segments for recoverable conversions

### ISSUE THREAD

**Issue by wiznillyp** [cdgriffith/FastFlix#243]
@cdgriffith,

Great software!  I wrote my own h.265 UHD conversion software before I ever heard of this, but this is much better than mine!

However, I tend to only convert overnight, so I like to have spots in the conversion that are easy to break and recover from. Is there any feature like this in this project?

FYI, To do this in my implementation, I first segment the source file into 3 minute video-only chunks (and a list) with the following parameters in the output string:

`-f segment -segment_time_delta 5 -segment_time 180 -segment_list <List_name>`

Then, I loop through the resulting files converting them one at a time and finally concatenating all of the segments back together using the segment list.

I can help implementing this here if it is useful, but I am not sure exactly how to fit it into your UI, especially since this should be able to work in between sessions. This also might help with #227

**Comment by cdgriffith** [comment]
I am very interested in adding something that could do the batch processing or simple concating as well like https://github.com/cdgriffith/FastFlix/issues/115 ask. Really haven't spent a lot of time on it yet, and really haven't gotten inspiration on how to throw it in without large changes. 

If you get ideas, please feel free to add them here!

**Comment by forart** [comment]
@wiznillyp There are also other interesting approaches - such as [Av1an](https://github.com/cdgriffith/FastFlix/issues/124) one - but needs to be "reimplemented" specifically for FF...

**Comment by forart** [comment]
Well, there are different approaches to video splitting for encoding purposes, so here's some that hopefully may help in its implementation (random order):

- @ValentinMastro's [AV1_SPLIT_ENCODE](https://github.com/ValentinMastro/av1_split_encode): 
_"First, we run the first pass on the whole source file and use the resulting log file to know which frames are keyframes. Then we cut the source file in splits on each keyframes. They are written in the RAM with the magicyuv codec (lossless, fast encoding/decoding). We run the second pass on each splits in parallel using as many threads as the user provides."_

- @martindisch's [shepherd](https://github.com/martindisch/shepherd): 
_1. Creates a temporary directory in your home directory._
_2. Extracts the audio and encodes it. This is not parallelized, but the time this takes is negligible compared to the video anyway._
_3. Splits the video into chunks. This can take relatively long, since you're basically writing the full file to disk again. It would be nice if we could read chunks of the file and directly transfer them to the hosts, but that might be tricky with ffmpeg._
_4. Spawns a manager and an encoder thread for every host. The manager creates a temporary directory in the home directory of the remote and makes sure that the encoder always has something to encode. It will transfer a chunk, give it to the encoder to work on and meanwhile transfer another chunk, so the encoder can start directly with that once it's done, without wasting any time. But it will keep at most one chunk in reserve, to prevent the case where a slow machine takes too many chunks and is the only one still encoding while the faster ones are already done._
_5. When an encoder is done and there are no more chunks to work on, it will quit and the manager transfers the encoded chunks back before terminating itself._
_6. Once all encoded chunks have arrived, they're concatenated and the audio stream added._
_7. All remote and the local temporary directory are removed._

- @nashspence's [The Video Split Script](https://github.com/nashspence/video-detect-visual-scenes-splitter):
_A script written in typescript that will use ffmpeg to split a video file into multiple clips based off of either the chapter marker embedded into the video file or an analysis and detection of a visual hard cut._

- @mrbeni's [PySceneDetect](https://github.com/mrbeni/sceneencoder):
_a command-line tool, written in Python and using OpenCV, which analyzes a video, looking for scene changes or cuts. The output timecodes can then be used with another tool (e.g. mkvmerge, ffmpeg) to split the video into individual clips. A frame-by-frame analysis can also be generated for a video, to help with determining optimal threshold values or detecting patterns/other analysis methods for a particular video._

- @slhck's [Scenecut Extractor](https://github.com/slhck/scenecut-extractor):
_This tool uses the [select filter](http://ffmpeg.org/ffmpeg-filters.html#select_002c-aselect) from ffmpeg to determine the scene cut probability of adjacent frames, and allows users to determine which frames (or at which timestamps) the scene cuts happen._

Other:
- @samuelcalegari's [FFMpeg-change-scene](https://github.com/samuelcalegari/ffmpeg-change-scene): _Scene Change Detection with Node.js and FFmpeg_
- @kieranjol's [scenechange](https://github.com/kieranjol/scenechange): _Insert chapter markers for scene changes using Matroska using ffmpeg and mkvpropedit_
- @geluso's [ffmpeg-scene-extraction](https://github.com/geluso/ffmpeg-scene-extraction): _automatically split a large movie file into component scenes_
- @bredd's [FVDetectScenes](https://github.com/bredd/FVDetectScenes): _Family Video Scene Detection Tool uses FFMpeg_

Hope that inspires !
